Guide/routing.markdown

```diff
@@ -0,0 +1,190 @@
+# Controller & Actions
+
+```toc
+```
+
+## Routing Basics
+
+In your project routes are defined in the `Web/Routes.hs`. Additionally to the defining a route, it also has to be added in `Web/FrontController.hs` to be picked up by the routing system.
+
+The simplest way to define a route is by using `AutoRoute`, which automatically maps each controller action to an url. For a `PostsController`, the definition in `Web/Routes.hs` will look like this:
+
+```haskell
+instance AutoRoute PostsController
+type instance ModelControllerMap WebApplication Post = PostsController
+```
+
+Afterwards enable the routes for `PostsController` in `Web/FrontController.hs` like this:
+
+```haskell
+instance FrontController WebApplication where
+    controllers = 
+        [ -- ...
+        , parseRoute @PostsController
+        ]
+```
+
+Now you can open e.g. `/Posts` to access the `PostsAction`.
+
+## Url Generation
+
+Use `pathTo` to generate a path to a given action:
+
+```haskell
+pathTo ShowPostAction { postId = "adddfb12-da34-44ef-a743-797e54ce3786" }
+-- /ShowPost?postId=adddfb12-da34-44ef-a743-797e54ce3786
+```
+
+To generate a full url, use `urlTo`:
+
+```haskell
+urlTo NewUserAction
+-- http://localhost:8000/NewUser
+```
+
+## AutoRoute
+
+Let's say our `PostsController` is define in `Web/Types.hs` like this:
+
+```haskell
+data PostsController
+    = PostsAction
+    | NewPostAction
+    | ShowPostAction { postId :: !(Id Post) }
+    | CreatePostAction
+    | EditPostAction { postId :: !(Id Post) }
+    | UpdatePostAction { postId :: !(Id Post) }
+    | DeletePostAction { postId :: !(Id Post) }
+```
+
+Using `instance AutoRoute PostsController` will give us the following routing:
+
+```haskell
+GET /Posts                          => PostsAction
+GET /NewPost                        => NewPostAction
+GET /ShowPost?postId={postId}       => ShowPostAction { postId }
+POST /CreatePost                    => CreatePostAction
+GET /EditPost?postId={postId}       => EditPostAction { postId }
+POST /UpdatePost?postId={postId}    => UpdatePostAction { postId }
+PATCH /UpdatePost?postId={postId}   => UpdatePostAction { postId }
+DELETE /DeletePost?postId={postId}  => DeletePostAction { postId }
+```
+
+The urls are very close to the actual action which is called. Action parameters are taken automatically from the request query. This design helps you to always know which action is actually called, when requesting an url. 
+
+### AutoRoute & Beautiful Urls
+
+Lots of modern browser don't even show the full url bar anymore (e.g. Safari and most mobile browsers). Therefore AutoRoute doesn't aim to generate the "most" beautiful urls out of the box. It's rather optimized for the needs of developers. If you need beautiful urls for SEO reasons, instead of using AutoRoute you can use the more manual APIs of IHP Routing. 
```

Comment:
Needs a link to the relevant section in the docs, currently the section `Custom Routing` on this file

Reply:
Fixed, good catch :)